code/modules/assembly/holder.dm

```diff
@@ -118,40 +111,29 @@
 	if(..())
 		return TRUE
 	to_chat(user, span_notice("You disassemble [src]!"))
-	if(a_left)
-		a_left.on_detach()
-		a_left = null
-	if(a_right)
-		a_right.on_detach()
-		a_right = null
+	for(var/obj/item/assembly/assembly in assemblies)
+		assembly.on_detach()
+		LAZYREMOVE(assemblies,assembly)
 	qdel(src)
 	return TRUE
 
 /obj/item/assembly_holder/attack_self(mob/user)
 	src.add_fingerprint(user)
-	if(!a_left || !a_right)
+	if(LAZYLEN(assemblies)==1)
 		to_chat(user, span_danger("Assembly part missing!"))
 		return
-	if(istype(a_left,a_right.type))//If they are the same type it causes issues due to window code
-		switch(tgui_alert(usr,"Which side would you like to use?",,list("Left","Right")))
-			if("Left")
-				a_left.attack_self(user)
-			if("Right")
-				a_right.attack_self(user)
-		return
-	else
-		a_left.attack_self(user)
-		a_right.attack_self(user)
+
+	for(var/obj/item/assembly/assembly in assemblies)
+		assembly.attack_self(user)
 
 
 /obj/item/assembly_holder/proc/process_activation(obj/D, normal = 1, special = 1)
 	if(!D)
 		return FALSE
```

Comment:
While you're here, replace these with better variable names, 1 with TRUE, and auto-doc this. It's kind of confusing.